services/api/src/messaging/players.service.ts

```diff
@@ -152,20 +154,31 @@ export const playerConnected: Handler = (
     role: newPlayerRole,
     status: gameService.playerHasMove(ws.roomId, newPlayerId)
       ? PlayerStatus.FigurePlaced
-      : PlayerStatus.ActionNotTaken,
+      : gameService.playerHasSkipped(ws.roomId, newPlayerId)
+        ? PlayerStatus.MoveSkipped
+        :PlayerStatus.ActionNotTaken,
   };
 
   successfullyJoined(ws, newPlayerId);
   subscribe(ws, newPlayer);
 
   if (newPlayer.status !== PlayerStatus.ActionNotTaken) {
-    const myTurn = gameService.findMoveByPlayerId(ws.roomId, newPlayerId);
     const players = getPlayers(ws.roomId);
 
-    if (myTurn) {
+    if (newPlayer.status === PlayerStatus.FigurePlaced) {
```

Comment:
We cannot leave publishing final board in ```(newPlayer.status !== PlayerStatus.ActionNotTaken)``` because if the spectator relogs/refreshes the game, when its completed, he will not see the final board